watchdog/watchdog.go

```diff
@@ -22,12 +22,36 @@ const (
 )
 
 // Program is a program at path with arguments
-type Program struct {
+type ProgramNormal struct {
 	Path   string
 	Args   []string
 	ExitOn ExitOn
 }
 
+func (p *ProgramNormal) GetPath() string {
```

Comment:
The `Get` pattern isn't idiomatic in go...

You should make the args for `program` private and remove `Get` from methods.

For example:

```go
type program struct {
  path string
  ...
}

  func (p program) Path() string {
    return p.path
  }
```